synapse/federation/federation_server.py

```diff
@@ -739,22 +739,20 @@ async def _handle_received_pdu(self, origin: str, pdu: EventBase) -> None:
 
         await self.handler.on_receive_pdu(origin, pdu, sent_to_us_directly=True)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return "<ReplicationLayer(%s)>" % self.server_name
 
     async def exchange_third_party_invite(
         self, sender_user_id: str, target_user_id: str, room_id: str, signed: Dict
-    ):
-        ret = await self.handler.exchange_third_party_invite(
+    ) -> None:
+        await self.handler.exchange_third_party_invite(
             sender_user_id, target_user_id, room_id, signed
         )
-        return ret
 
-    async def on_exchange_third_party_invite_request(self, event_dict: Dict):
-        ret = await self.handler.on_exchange_third_party_invite_request(event_dict)
-        return ret
+    async def on_exchange_third_party_invite_request(self, event_dict: Dict) -> None:
+        await self.handler.on_exchange_third_party_invite_request(event_dict)
```

Comment:
this method doesnt return results anymore?

Reply:
Hmm...so this is a bit tricky `FederationHandler.on_exchange_third_party_invite_request` always returns `None`, but it looks like the result of this is actually returned via `FederationThirdPartyInviteExchangeServlet`, so something is a bit funky here.

It looks like it *should* be just returning `{}` instead.

Reply:
Oops, actually `FederationThirdPartyInviteExchangeServlet` calls the handler, not here so that's unrelated.

Reply:
I'm just pointing to changing the method behaviour on this, i have no other context, but no worries.

Reply:
7e030a2674c5e903f84c4e9c3335c5524b35b415 will stop using the returned value from this, which is always `None`.